Server/Source/AudioWorker.hpp

```diff
@@ -103,18 +103,10 @@ class AudioWorker : public Thread, public LogTagDelegate {
     bool waitForData();
 
     template <typename T>
-    AudioBuffer<T>* getProcBuffer();
-
-    template <>
-    AudioBuffer<float>* getProcBuffer() {
+    AudioBuffer<T>* getProcBuffer() {
```

Comment:
Why was that change needed?

Reply:
because without that it was giving compilation error on Linux.
both the definitions were specializations and in class and MSVC accepts the previous code but gcc on linux is strict. 
we can proceed with specialization definition out of class  definition, but that in header file , gives multiple definition error at linking time.
I declared one as general template definition and moved other specialization in cpp file. 
Still only two definitions (for float and double) will be generated. Trying with any other type will give compiler error as default float*  type buffer return value,  will not match with intended type and require another explicit specialization.